### DISCUSSION TITLE
install without electron

### DISCUSSION THREAD

**Discussion by essential-existence** [jgraph/drawio#2408] [Q&A] [answered]
Hi. Can this be run on any server (e.g. node.js) like a normal web application, without any server and electron dependencies? I would like to setup this application in termux with node.js

**Comment by davidjgraph** [comment]
Like a normal web application without a server? How does that work? All web applications are served from a server.

**Reply by h1z1** [reply]
file:///path technically isn't  :)    FWIW I cannot stand electron either (nor anything like snap).

**Comment by davidjgraph** [comment]
Serving from local filesystem will block XHR calls.

**Reply by drawfund-founder** [reply]
termux & termux-setup-storage

```bash
pkg install nodejs
npm install -g http-server
cd $home
cd storage/shared/1rd-Termux/drawio/src/main/webapp
http-server
```
http://127.0.0.1:8080/

**Comment by essential-existence** [comment, marked as answer]
Yes, I managed to run it on node.js server. Only some files are not found, since they are requested, for example, along the path `/images/android-chrome-196x196.png`, and not, for example, `/drawio/images/google-drive-logo.svg`.
By "regular server-independent web application" I mean a static web application where the server only serves static files.
Thanks to all!

**Reply by davidjgraph** [reply]
Yes, you could in theory serve only the JS. You wouldn't be able to authenicate to any cloud providers. PDF export won't work, you could print to PDF.

**Comment by editfund-founder** [comment]
> Install Java and Tomcat in Termux.
> Copy draw.war to Tomcat.
```shell
~/tomcat/bin $ chmod 777 catalina.sh
~/tomcat/bin $ sh startup.sh
Using CATALINA_BASE:   /home/tomcat
Using CATALINA_HOME:   /home/tomcat
Using CATALINA_TMPDIR: /home/tomcat/temp
Using JRE_HOME:        /data/data/com.termux/files/usr
Using CLASSPATH:       /home/tomcat/bin/bootstrap.jar:/home/tomcat/bin/tomcat-juli.jar
Using CATALINA_OPTS:
Tomcat started.
~/tomcat/bin $
```
